src/lib/bindReporter.ts

```diff
@@ -17,23 +17,38 @@
 import {Metric, ReportCallback} from '../types.js';
 
 
+const getRating = (value: number, thresholds: number[]) => {
+  if (value > thresholds[1]) {
+    return 'poor';
+  }
+  if (value > thresholds[0]) {
+    return 'ni';
+  }
+  return 'good';
+};
```

Comment:
Should we check its >= 0 before returning `good`?

Maybe return null or such like if < 0

Reply:
I don't think we have to. This function is only ever called from within a conditional that check if `metric.value >= 0` (see line 40).

Reply:
Seems a good safety check to add in case that ever becomes not there case?

Reply:
I don't want to type that it could be `string | null`, because I think that would unnecessarily concern users of the library (especially TypeScript users).

I think to handle your concern it's better to have a test that checks that case rather than building it into the code (that users have to download even though it's not currently needed). There are lots of tests currently that make sure the metric value is greater than or equal to 0, and those would all start failing if that were ever not the case.

Reply:
Ah typescript! Must learn that…

Not a big deal, so fine to resolve this. Plus arguably a negative number IS a “good” score by the metric definitions 😄And the bug if that ever happens is upstream not in this function.

Alternatively could also return “unknown”.